plugins/in_http/http_prot.c

```diff
@@ -510,21 +526,71 @@ static int process_payload(struct flb_http *ctx, struct http_conn *conn,
         type = HTTP_CONTENT_URLENCODED;
     }
 
+    /* handle gzip-encoded content */
+    /* 
+        The way things come out of headers_extra is very strange; everything seems to be in
+        key->data of the first entry and val->data is everything minus the first header key.
+        The behavior of monkey seems to be pretty buggy here.
+
+        ...so, we just grab the whole thing and then sift through it looking for content-encoding.
+    */
+    if (session->parser.headers_extra_count > 0) {
+        extra_headers = &session->parser.headers_extra[0];
+        const char *content_encoding = "content-encoding";
+        if (extra_headers->key.data != NULL) {
+            extracted_extra_headers = mk_string_tolower(extra_headers->key.data);
+            content_encoding_header_location = mk_string_search(extracted_extra_headers, content_encoding, 0);
+        }
+    }
```

Comment:
It seems strange. In your environment, like https://github.com/fluent/fluent-bit/blob/master/plugins/in_splunk/splunk_prot.c#L509-L522 operation should not work as expected?